### PR TITLE
Return the function trigger URL in the API

### PR DESCRIPTION
**Pull request by davidebbo**

### CODE REVIEW

File: Kudu.Core/Functions/FunctionManager.cs [hunk 1]
```diff
@@ -191,6 +192,17 @@ private FunctionEnvelope CreateFunctionConfig(string configContent, string funct
             };
         }
 
+        private string GetFunctionTriggerUrl(string functionName)
+        {
+            FunctionSecrets functionSecrets = GetFunctionSecrets(functionName);
+
+            // TODO: 'code' is only valid for WebHooks. Need to generalize
+            return String.Format(@"https://{0}/api/{1}?code={2}",
+                System.Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME") ?? "localhost",
```
Comment (ahmelsayed) [inline Kudu.Core/Functions/FunctionManager.cs]: any particular reason for using this over `_environment.AppBaseUrlPrefix`?

File: Kudu.Core/Functions/FunctionManager.cs [hunk 2]
```diff
@@ -284,9 +296,40 @@ private string GetFunctionSampleDataFilePath(string functionName)
...
         }
 
+        private FunctionSecrets GetFunctionSecrets(string functionName)
+        {
+            FunctionSecrets secrets;
+            string secretFilePath = GetFunctionSecretsFilePath(functionName);
+            if (File.Exists(secretFilePath))
+            {
+                // load the secrets file
+                string secretsJson = File.ReadAllText(secretFilePath);
```
Comment (ahmelsayed) [inline Kudu.Core/Functions/FunctionManager.cs]: use `FileSystemHelpers.ReadAllTextFromFileAsync`

File: Kudu.Core/Functions/FunctionManager.cs [hunk 3]
```diff
@@ -284,9 +296,40 @@ private string GetFunctionSampleDataFilePath(string functionName)
...
+            else
+            {
+                // initialize with new secrets and save it
+                secrets = new FunctionSecrets
+                {
+                    Key = SecurityUtility.GenerateSecretString()
+                };
+
+                FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(secretFilePath));
+                File.WriteAllText(secretFilePath, JsonConvert.SerializeObject(secrets, Formatting.Indented));
```
Comment (ahmelsayed) [inline Kudu.Core/Functions/FunctionManager.cs]: `FileSystemHelpers.WriteAllTextFromFileAsync`

File: Kudu.Core/Functions/FunctionManager.cs (hunk 1 above)
Comment (davidebbo) [inline Kudu.Core/Functions/FunctionManager.cs]: scm vs main site

**Pull request merged**